irc/handlers.go

```diff
@@ -194,20 +229,23 @@ func sendSuccessfulRegResponse(client *Client, rb *ResponseBuffer, forNS bool) {
 	if forNS {
 		rb.Notice(client.t("Account created"))
 	} else {
-		rb.Add(nil, client.server.name, RPL_REGISTRATION_SUCCESS, client.nick, client.AccountName(), client.t("Account created"))
+		rb.Add(nil, client.server.name, RPL_REG_SUCCESS, client.nick, client.AccountName(), client.t("Account created"))
 	}
-	sendSuccessfulSaslAuth(client, rb, forNS)
+	sendSuccessfulAccountAuth(client, rb, forNS, false)
 }
 
-// sendSuccessfulSaslAuth means that a SASL auth attempt completed successfully, and is used to dispatch messages.
-func sendSuccessfulSaslAuth(client *Client, rb *ResponseBuffer, forNS bool) {
+// sendSuccessfulAccountAuth means that an account auth attempt completed successfully, and is used to dispatch messages.
+func sendSuccessfulAccountAuth(client *Client, rb *ResponseBuffer, forNS, forSASL bool) {
```

Comment:
`forNS` and `forSASL` are an XOR, right? Should this just be a single bool?

Reply:
Mmm. I have thoughts about this. `forSASL` is sort of a dodgy name, it's basically trying to capture the case of 'for login' vs 'for ... um, not logging in, like registration?' since those are the two places we use that function. `forSASL`'s just the easiest way to present it, so long as we assume that NS identify is also 'for sasl'.